### PR TITLE
Add CustomPopper and Add TableFiltersMenu to PaymentsList

### LINKED ISSUE #720: Implement Dashboard Payments filters to Payments List component
Create custom-popper component and create filters menu using custom-popper. 

Implement new filters-menu component to payments-list

### PR DESCRIPTION
**Pull request by colganmeanor**
This PR commits the following changes:

- Adds CustomPopper component for internal use. 
  - Uses Popper.js directly instead of trying to use bootstrap JS
  - contained in the shadow dom of its parent components
- Adds TableFiltersMenu component which uses CustomPopper
- Creates PaymentsListFilters component which renders TableFiltersMenu along with form inputs
- Render PaymentsListFilters component in PaymentsList 

Links
-----

Closes #720 

Developer considerations
--------------

- On any task
  - [ ] Previous unit and e2e tests are passing
  - [ ] Perform dev QA on Chrome, Firefox, and Safari
- On new features
  - [ ] Add unit tests
  - [ ] Add e2e tests
  - [ ] Add documentation
  - [ ] Does the component have exportedparts for styling? If so, add unit a unit test for each exportedpart
- On bugs
  - [ ] Add unit tests to prevent the bug from happening again

Developer QA steps
--------------------

Let's mainly focus QA on the implementation of the CustomPopper component into the PaymentsList as the new TableFiltersMenu

- [x] Menu can be successfully opened with a mouse click on the Filters button
- [x] Menu can be successfully closed by clicking on the Filter button while the menu is open, or clicking anywhere else on the screen. 
- [x] Terminal ID filter input successfully applies `terminal_id` param
- [x] Payment Status filter input successfully applies `payment_status` param
- [x] Start Date filter input successfully applies `created_after` param
- [x] End Date filter input successfully applies `created_before` param
- [x] When the filter menu is used to set a filter - the Clear Filters button should be rendered
- [x] Clear Filters button resets params to {}
- [x] Pagination next or pagination previous does NOT render the Clear Filters button
- [x] Filter Menu can be used in conjunction with pagination - IE set payment status to Failed and can still successfully tab through next and previous pages.

### CODE REVIEW

File: apps/docs/vite.config.ts [hunk 1]
```diff
@@ -3,4 +3,7 @@ import react from '@vitejs/plugin-react';
 
 export default defineConfig({
   plugins: [react()],
+  optimizeDeps: {
```
Comment (TiagoFuelber) [inline apps/docs/vite.config.ts:6]: Interesting, I'm curious about this.

File: packages/webcomponents/src/ui-components/custom-popper/custom-popper.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      this.popperContentRef.removeEventListener('mouseleave', this.hide.bind(this));
+    } else if (this.trigger === 'focus') {
+      this.anchorRef.removeEventListener('focus', this.show.bind(this));
+      this.anchorRef.removeEventListener('blur', this.hide.bind(this));
+    }
+  }
+
+  @Listen('click', { target: 'window' })
+  handleClick(event: MouseEvent) {
+    const path = event.composedPath();
```
Comment (TiagoFuelber) [inline packages/webcomponents/src/ui-components/custom-popper/custom-popper.tsx:51]: Nice!

**Review by TiagoFuelber — DISMISSED** [review]
Awesome work! 💯

File: apps/docs/vite.config.ts (hunk 1 above)
Comment (colganmeanor) [reply apps/docs/vite.config.ts]: Ah yeah I removed this in my recent push. Thanks for pointing that out. I was having local issues with storybook cache and was experimenting with vite's config file to fix the issue. I forgot I had included that in this PR. It doesn't do anything, so i removed it!

**Review by jakemer — APPROVED** [review]
This turned out really nice! Good work!

**Pull request merged**